src/Avalonia.Visuals/Matrix.cs

```diff
@@ -54,7 +56,7 @@ namespace Avalonia
         /// <summary>
         /// HasInverse Property - returns true if this matrix is invertible, false otherwise.
         /// </summary>
-        public bool HasInverse => GetDeterminant() != 0;
+        public bool HasInverse => Math.Abs(GetDeterminant()) >= double.Epsilon;
```

Comment:
Unfortunately `double.Epsilon` is not the value that we would like
> Note
The value of the Epsilon property is not equivalent to machine epsilon, which represents the upper bound of the relative error due to rounding in floating-point arithmetic.

It would be nice to see helper here to define one constant everywhere. 

It seems to me that single precision is sufficient, because in graphics it is usually used, since there is not much accuracy needed, and the performance of operations on them is higher. But maybe in Avalonia the matrix is ​​used for something else and therefore double precision is important

Reply:
This is a good point, I guess comparison there should be using https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Base/Utilities/MathUtilities.cs#L89 which was ported from WPF with a wrong constant https://github.com/dotnet/wpf/blob/master/src/Microsoft.DotNet.Wpf/src/Shared/MS/Internal/DoubleUtil.cs#L174.

I won’t have much time today so if you have a moment you are more than welcome to open a PR correcting our utility function :)

Reply:
Switched to using `MathUtilities.IsZero`.